### PR TITLE
Fix CSRF

### PR DESCRIPTION
**Pull request by cgrothaus**
# Goal
Fix CSRF issue. On a current Phoenix (1.6.6) with Ueberauth (0.7.0) project, using `ueberauth_oidc` runs into a CSRF failure. 

# Changes
Ueberauth provides the CSRF token with a `state` param. This param needs to be passed on to the OpenID connect provider in the _request_ phase, so that in the _callback_ phase it is part of the redirect back from the OIDC provider to our strategy.

### CODE REVIEW

File: lib/ueberauth/strategy/oidc.ex [hunk 1]
```diff
@@ -14,7 +14,9 @@ defmodule Ueberauth.Strategy.OIDC do
   """
   def handle_request!(conn) do
     provider_id = conn |> get_options!() |> get_provider()
+
     params = params_from_conn(conn)
+    params = with_state_param([], conn) |> Enum.into(params)
```
Comment (Kuret) [inline lib/ueberauth/strategy/oidc.ex]: Could use the existing `params_from_conn()` for this, and directly get the `conn.private[:ueberauth_state_param]
` over there and add it to the returned map.

We need to add a test to check if the state is correctly retrieved too.

Comment (cgrothaus) [reply lib/ueberauth/strategy/oidc.ex]: Yep. I changed it.

**Comment by Kuret** [comment]
~~Can you tell me a bit more about which scenario's you encounter the CSRF errors? Just tried an OIDC connection with one of our providers in a Phoenix 1.6 application and I am getting no such errors.~~ Ignore this, had the wrong `Ueberauth` version.

**Comment by cgrothaus** [comment]
Yeah, it's from the newest Ueberauth 0.7.0 (which strangely has no tag in Github, but is released on Hex). The error is raised in [Ueberauth.Strategy](https://github.com/ueberauth/ueberauth/blob/master/lib/ueberauth/strategy.ex#L351). This was added by https://github.com/ueberauth/ueberauth/pull/136.

**Comment by Kuret** [comment]
Reproduced the CSRF error with Ueberauth 0.7, and confirmed your PR fixes it. I'll push these changes to a new version on Hex.pm in a bit.

**Comment by cgrothaus** [comment]
Thanks for getting back to this and checking it 😍 .

**Comment by Kuret** [comment]
No problem, thanks for the fix!
I just published this as version `0.1.3` on Hex.pm btw.

**Pull request merged**